File: navigationinjector/src/main/java/com/gc/navigationinjector/State.java
package com.gc.navigationinjector;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface State {
    Class<? extends Operation> Operation();
    Class<? extends BaseViewModel> ViewModel();
    Class<? extends BaseActivity> Activity() default BaseActivity.Default.class;
    Class<? extends BaseFragment> Fragment() default BaseFragment.Default.class;
}

File: navigationinjector/src/main/java/com/gc/navigationinjector/State.kt
package com.gc.navigationinjector

import kotlin.reflect.KClass

/**
 * Created by 674756 on 05/02/2018.
 */
annotation class State(
        val Operation : KClass<out Operation>,
        val ViewModel : KClass<out BaseViewModel>,
        val Activity : KClass<out BaseActivity> = BaseActivity.Default :: class,
        val Fragment : KClass<out BaseFragment> = BaseFragment.Default :: class
)